### PR TITLE
perf(components): [el-scrollbar] prevent re-render when scrolling

### PR DESCRIPTION
**Pull request by msidolphin**
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/scrollbar/src/scrollbar.vue [hunk 1]
```diff
@@ -69,6 +69,7 @@ export default defineComponent({
 
     const sizeWidth = ref('0')
     const sizeHeight = ref('0')
+    const barRef = ref()
```
Comment (JeremyWuuuuu) [inline packages/components/scrollbar/src/scrollbar.vue:72]: Can we specify `barRef` type via

```ts
const barRef = ref<InstanceType<typeof<Bar>>()
```

File: packages/components/scrollbar/src/thumb.vue [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+  toRef,
+} from 'vue'
+import { useEventListener, isClient } from '@vueuse/core'
+import { scrollbarContextKey } from '@element-plus/tokens'
+import { throwError } from '@element-plus/utils/error'
+import { BAR_MAP, renderThumbStyle } from './util'
+
+import { thumbProps } from './thumb'
+
+const COMPONENT_NAME = 'Bar'
```
Comment (JeremyWuuuuu) [inline packages/components/scrollbar/src/thumb.vue:35]: **Question:** why `thumb` is named `Bar`?

**Review by jw-foss — COMMENTED** [review]
Most of your changes make sense to me, please address the comments I have left for you. :>

File: packages/components/scrollbar/src/thumb.vue [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+    const instance = ref<HTMLDivElement>()
+    const thumb = ref<HTMLDivElement>()
+
+    const barStore = ref({})
+    const visible = ref(false)
+
+    let cursorDown = false
+    let cursorLeave = false
+    let onselectstartStore:
```
[the remark this answers is not in the document]
Comment (msidolphin) [reply packages/components/scrollbar/src/thumb.vue:53]: I changed to `thumbState`, I think is more appropriate

File: packages/components/scrollbar/src/thumb.vue [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+
+  setup(props) {
+    const scrollbar = inject(scrollbarContextKey)
+    if (!scrollbar)
+      throwError(COMPONENT_NAME, 'can not inject scrollbar context')
+
+    const instance = ref<HTMLDivElement>()
+    const thumb = ref<HTMLDivElement>()
+
+    const barStore = ref({})
```
[the remark this answers is not in the document]
Comment (msidolphin) [reply packages/components/scrollbar/src/thumb.vue]: I changed to thumbState, I think is more appropriate

**Comment by msidolphin** [comment]
@JeremyWuuuuu done

**Review by jw-foss — APPROVED** [review]

**Pull request merged**